### PR TITLE
Migrate health in details

### PR DESCRIPTION
**Pull request by aljesusg**
Require: https://github.com/kiali/kiali/pull/4799
Related with: https://github.com/kiali/kiali/issues/4611

### CODE REVIEW

**Comment by aljesusg** [comment]
In workload details there are 2 more health requests I don't know if they are related with minigraph

**Comment by lucasponce** [comment]
As commented, it seems the refresh when duration changes are not well propagated.

**Comment by aljesusg** [comment]
> As commented, it seems the refresh when duration changes are not well propagated.

Fixed and ready for review ;)

**Review by lucasponce — APPROVED** [review]

**Comment by lucasponce** [comment]
Added @jshaughn https://github.com/kiali/kiali/pull/4799#issuecomment-1068959626

File: src/pages/AppDetails/AppDetailsPage.tsx [hunk 1]
```diff
@@ -12,23 +12,27 @@ import IstioMetricsContainer from '../../components/Metrics/IstioMetrics';
...
   // currentTab is needed to (un)mount tab components
   // when the tab is not rendered.
   currentTab: string;
 };
 
 type ReduxProps = {
   jaegerInfo?: JaegerInfo;
+  duration: DurationInSeconds;
   lastRefreshAt: TimeInMilliseconds;
   timeRange: TimeRange;
```
Comment (jshaughn) [inline src/pages/AppDetails/AppDetailsPage.tsx:37]: probably just me but I like to keep these lists alphabetized

File: src/pages/AppDetails/AppInfo.tsx [hunk 2]
```diff
@@ -10,21 +8,21 @@ import GraphDataSource from 'services/GraphDataSource';
...
 import MiniGraphCard from '../../components/CytoscapeGraph/MiniGraphCard';
 import { GraphEdgeTapEvent } from '../../components/CytoscapeGraph/CytoscapeGraph';
 import history, { URLParam } from '../../app/History';
 
 type AppInfoProps = {
   app?: App;
+  health?: AppHealth;
   duration: DurationInSeconds;
   lastRefreshAt: TimeInMilliseconds;
   mtlsEnabled: boolean;
```
Comment (jshaughn) [inline src/pages/AppDetails/AppInfo.tsx:22]: still just me, but it was alphabetized and now it's not :-)

**Review by hhovsepy — APPROVED** [review]
Verified from functionality testing and regression perspective.

File: src/pages/WorkloadDetails/WorkloadDetailsPage.tsx [hunk 3]
```diff
@@ -22,14 +23,17 @@ import WorkloadWizardDropdown from '../../components/IstioWizards/WorkloadWizard
...
   workload?: Workload;
+  health?: WorkloadHealth;
   currentTab: string;
 };
 
 type ReduxProps = {
   jaegerInfo?: JaegerInfo;
+  duration: DurationInSeconds;
   lastRefreshAt: TimeInMilliseconds;
   statusState: StatusState;
```
Comment (jshaughn) [inline src/pages/WorkloadDetails/WorkloadDetailsPage.tsx:38]: same :-)

File: src/pages/WorkloadDetails/WorkloadDetailsPage.tsx [hunk 4]
```diff
@@ -297,6 +315,7 @@ class WorkloadDetails extends React.Component<WorkloadDetailsPageProps, Workload
 
 const mapStateToProps = (state: KialiAppState) => ({
   jaegerInfo: state.jaegerState.info,
+  duration: durationSelector(state),
   lastRefreshAt: state.globalState.lastRefreshAt,
   statusState: state.statusState
```
Comment (jshaughn) [inline src/pages/WorkloadDetails/WorkloadDetailsPage.tsx:320]: same

**Review by jshaughn — APPROVED** [review]
Tested with https://github.com/kiali/kiali/pull/4799 and seems to be working.  Some non-blocking comments.

**Review by israel-hdez — DISMISSED** [review]
I'm Jay's follower on alphabetization :-) Not fully required since we want to move forward this, and because most likely somebody else will alphabetize later.

One variable needs renaming.

File: src/types/Health.ts [hunk 5]
```diff
@@ -47,6 +47,16 @@ export interface WorkloadStatus {
   syncedProxies: number;
 }
 
+export interface AppHealthResponse {
+  workloadStatus: WorkloadStatus[];
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply src/types/Health.ts:51]: This comes from the backend, the API return that naming...it's not related with this PR if we want to change it we must to make changes in amother views and in the backend

**Comment by aljesusg** [comment]
Ui: https://github.com/kiali/kiali-ui/pull/2334
Backend https://github.com/kiali/kiali/pull/4808

**Pull request merged**